src/Chainweb/Miner/Coordinator.hs

```diff
@@ -92,16 +86,17 @@ import Utils.Logging.Trace (trace)
 -- made as often as desired, without clogging the Pact queue.
 --
 newtype PrimedWork =
-    PrimedWork (HM.HashMap MinerId (HM.HashMap ChainId (Maybe CachedPayload)))
+    PrimedWork (HM.HashMap MinerId (HM.HashMap ChainId (Maybe PayloadWithOutputs)))
     deriving newtype (Semigroup, Monoid)
 
-type CachedPayload = T2 PayloadWithOutputs BlockCreationTime
-
 -- | Data shared between the mining threads represented by `newWork` and
 -- `publish`.
 --
+-- The key is a unique pair of `BlockHash` of the parent block with the hash of
+-- the current block's payload.
+--
 newtype MiningState = MiningState
-    (M.Map (T2 BlockCreationTime BlockPayloadHash) (T3 Miner PrevTime PayloadWithOutputs))
+    (M.Map (T2 BlockHash BlockPayloadHash) (T3 Miner PayloadWithOutputs (Time Micros)))
```

Comment:
@emilypi and I discussed this, and we believe that this pair is a sufficiently unique key for cached payloads.